woocommerce/woocommerce-ios
# Adds Storage layer for Customer and CustomerSearchResult

Part of: #7741 
Branched from https://github.com/woocommerce/woocommerce-ios/pull/7824 , so that PR should be merged first.

### Description
This PR adds the Storage Layer for `Customer`, and `CustomerSearchResult` and  by updating our Data Model with both entities, as well as their respective properties.

### Changes
- Updated Data Model from 73 to 74.
- Added `Customer`, and `CustomerSearchResult` entities.
- Generated `+CoreDataClass` and `+CoreDataProperties` for both entities.
- Added a `73_to_74` test to MigrationTests to confirm that the new entity and properties can be found after migration.
- Updated MIGRATIONS.md

### Testing instructions
- Unit Tests should pass
- Attempt to run the app and check that doesn't crash :D
